src/explore-education-statistics-common/src/components/form/FormComboBox.tsx

```diff
@@ -156,158 +158,210 @@ const FormComboBox = ({
+                  case 'ArrowRight': {
+                    const directionChange = event.key === 'ArrowLeft' ? -1 : 1;
+
+                    inputEl.selectionStart = inputEl.selectionStart
+                      ? inputEl.selectionStart + directionChange
+                      : 0;
+                    inputEl.selectionEnd = inputEl.selectionStart;
+                    inputEl.focus();
+                    break;
                   }
+
+                  case 'Enter':
+                    onSelect(selectedOption);
+                    toggleShowOptions(false);
+                    break;
+
+                  case 'Home':
+                    inputEl.selectionStart = 0;
+                    inputEl.selectionEnd = 0;
+                    inputEl.focus();
+                    break;
+
+                  case 'End':
+                    inputEl.selectionStart = inputEl.value.length;
+                    inputEl.selectionEnd = inputEl.selectionStart;
+                    inputEl.focus();
+                    break;
+
+                  case 'Escape':
+                    setValue('');
+                    toggleShowOptions(false);
+                    inputEl.focus();
+                    break;
+
+                  default:
                 }
-              }}
-            >
-              {renderedOptions &&
-                renderedOptions.length > 0 &&
-                renderedOptions.map((item, index) => {
-                  const key = index;
-
-                  return (
-                    // eslint-disable-next-line jsx-a11y/click-events-have-key-events
-                    <li
-                      aria-selected={selectedOption === index}
-                      key={key}
-                      id={`${id}-option-${index}`}
-                      className={
-                        selectedOption === index ? styles.selected : undefined
-                      }
-                      role="option"
-                      ref={el => {
-                        if (el) {
+              }
+            }}
+          >
+            {renderedOptions &&
+              renderedOptions.length > 0 &&
+              renderedOptions.map((item, index) => {
+                const key = index;
+
+                return (
+                  // eslint-disable-next-line jsx-a11y/click-events-have-key-events
+                  <li
+                    aria-selected={selectedOption === index}
+                    key={key}
+                    id={`${id}-option-${index}`}
+                    aria-setsize={renderedOptions.length}
+                    aria-posinset={index + 1}
+                    className={
+                      selectedOption === index ? styles.selected : undefined
+                    }
+                    role="option"
+                    tabIndex={-1}
+                    ref={el => {
+                      if (el) {
+                        if (selectedOption === index) {
+                          el.focus();
```

Comment:
This is a smell - we shouldn't be focusing elements like this. Refs shouldn't be used for side effects.

Instead, we should do something like `PublicationPublishedReleasesTable` which refocuses whenever a `focusReleaseId` changes. In our case, it'd look something like:

```tsx
const optionRefs = useRef<Dictionary<HTMLLIElement>>({});

useEffect(() => {
  if (selectedOption) {
    optionRefs.current[selectedOption]?.focus();
  }
}, [selectedOption]);


// Revert the `ref` back to what it was previously

<li
  ref={el => {
    if (el) {
      optionRefs.current[key] = el;
    }
  }}
>
```

Reply:
Had a look at this & it seems for some reason that the selectedOption gets set twice when doing this which results in two li elements sometimes having the `selected` class at the same time. In addition to that, the `useEffect` doesn't seem to be setting the focus on the list item instead & the focus stays on the input. For now, I've done the following which is a bit hacky but seems to work. In the future, it might be worth re-working this component to get around this problem.

```typescript 
 ref={el => {
                      if (el) {
                        optionRefs.current[index] = el;

                        if (isSelected) {
                          // this is a bit hacky but it seems to be the only way to focus
                          // without two list items being selected at the same time
                          // (and the focus being on the input instead of the list item)
                          el.focus();
                        }
                      }
                    }}

```